CONTRIBUTING.md

```diff
@@ -2,6 +2,11 @@
 Contributing to scikit-learn
 ============================
 
+The latest contributing guide is available in the repository at
+`doc/developers/contributing.rst`, or online at:
+
+https://scikit-learn.org/dev/developers/contributing.html
```

Comment:
I've been thinking about (and maybe we had some discussion about it?) having the link from the stable website to the dev for the contributing guide. But I'm more than happy with this note here.

Reply:
I think this note is good emough.